box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFile.java

```diff
@@ -1206,4 +1219,443 @@ public void setRequestedPage(int page) {
+     */
+    public static class CreateUploadSession extends BoxRequest<BoxUploadSession, CreateUploadSession> {
+        private static final long serialVersionUID = 8145675031279971502L;
+
+        private String mDestinationFolderId;
+        private String mFileName;
+        private long mFileSize;
+        private FileInputStream mFileInputStream;
+
+        /**
+         * Creates an upload file session  with the default parameters
+         *
+         * @param file file to upload
+         * @param destinationFolderId   id of the parent folder for the new file
+         * @param requestUrl    URL of the upload file endpoint
+         * @param session   the authenticated session that will be used to make the request with
+         */
+        public CreateUploadSession(File file, String destinationFolderId, String requestUrl, BoxSession session)
+                throws FileNotFoundException {
+            super(BoxUploadSession.class, requestUrl, session);
+            mRequestUrlString = requestUrl;
+            mRequestMethod = Methods.POST;
+            mFileName = file.getName();
+            mFileSize = file.length();
+            mFileInputStream = new FileInputStream(file);
+            mDestinationFolderId = destinationFolderId;
+            mBodyMap.put("folder_id", destinationFolderId);
+            mBodyMap.put("file_size", mFileSize);
+            mBodyMap.put("file_name", mFileName);
+        }
+
+        @Override
+        protected void onSendCompleted(BoxResponse<BoxUploadSession> response) throws BoxException {
+            if (response.isSuccess()) {
+                BoxUploadSession uploadSession = response.getResult();
+
+                try {
+                  computeSha1s(mFileInputStream, uploadSession, mFileSize);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new BoxException("Can't compute sha1 for file",e);
+                } catch (IOException e) {
+                    throw new BoxException("Can't compute sha1 for file",e);
+                }
+            }
+            super.onSendCompleted(response);
+        }
+
+
+
+        //Pre-compute sha1s for sending in future calls and save them to BoxUploadSession
+        static void computeSha1s(FileInputStream fileInputStream, BoxUploadSession uploadSession,
+                                 long fileSize)
+                throws NoSuchAlgorithmException, IOException {
+            int totalParts = uploadSession.getTotalParts();
+            List<String> partSha1s = new ArrayList<>(totalParts);
+
+
+            byte[] partBuffer = new byte[uploadSession.getPartSize()];
+
+            MessageDigest mdFile = MessageDigest.getInstance("SHA-1"); //Store sha1 over entire file
+            MessageDigest mdPart = MessageDigest.getInstance("SHA-1");
+            for (int i = 0; i < totalParts; i++) {
+                partBuffer = new byte[BoxUploadSession.getChunkSize(uploadSession, i, fileSize)];
```

Comment:
Here we can have the same memory issue I mentioned before. The chunk size can be 8mb or even more (depending on the file total size). That amount of data in RAM might be risky for mobile devices.
I see that the digest API does not take streams, so can we allocate a smaller byte array size (lets say 16Kbytes) and update the digest in a loop ?